### ISSUE TITLE
SC2185 find shows an error when using -O3 despite there being a path

### ISSUE THREAD

**Issue by samcv** [koalaman/shellcheck#942]
#### For bugs
- Rule Id SC2185
- My shellcheck version 0.4.6
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
find -O3 ~/git
```

#### Here's what shellcheck currently says:

Some finds don't have a default path. Specify '.' explicitly. [SC2185]

#### Here's what I wanted or expected to see:

Should have no error. When using either -O1 -O2 -O3, the optimization level must come before the directory. Shellcheck should make sure the if -Ox is provided that it checks the term after it for a directory there.

**Comment by TorgeH** [comment]
seems to be the same for "-D" debug levels.
BTW, wouldn't it be a good idea to have deep-links to examples such that they could be added to the error reports?

**Comment by koalaman** [comment]
Indeed. 

Various implementations have various pre-path flags. POSIX has `-L` and `-H`. GNU has `-P`. macOS has `-X`.  Instead of trying to enumerating them all, ShellCheck got along by assuming they were all single characters. 

`-O3` is 2, like `-ok`, so it was incorrectly considered a predicate by this naive matcher. This case has been explicitly added, improving the situation. Thanks!

For the existing version, there's no good rewrite, so please disable the message with a directive.
